### DISCUSSION TITLE
Question: Cucumber hooks execution order

### DISCUSSION THREAD

**Discussion by m4rtelli** [cucumber/common#2221] [Q&A JavaScript] [answered]
I recently upgraded my project to support Cucumber 11.0.1 and I've noticed that my After() hooks are now executing in a top->down order. I recall Before() executed this way but After() hooks typically executed in reverse order. Has this changed with version 11+? My sincerest apologies if this was documented somewhere. I couldn't locate it. 

Thanks a bunch for your time.

**Comment by mpkorstanje** [comment]
https://github.com/cucumber/cucumber-js/blob/f0122a3dc75a5ef0bcce0d0cc62565aecfbc92a0/docs/support_files/hooks.md#hooks
> Hooks are used for setup and teardown the environment before and after each scenario. See the [API reference](https://github.com/cucumber/cucumber-js/blob/f0122a3dc75a5ef0bcce0d0cc62565aecfbc92a0/docs/support_files/api_reference.md) for the specification of the first argument passed to hooks. Multiple Before hooks are executed in the order that they were defined. Multiple After hooks are executed in the reverse order that they were defined.

So that sounds like a bug. You'll want to report it in https://github.com/cucumber/cucumber-js.

**Comment by davidjgoss** [comment, marked as answer]
Dealt with in https://github.com/cucumber/cucumber-js/issues/2455
